src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -2142,67 +2141,75 @@ class module_sram : public module_impl
+        (xrt_core::dlsym(dtrace.lib_hdl.get(), "get_dtrace_col_numbers"));
+    if (!get_dtrace_col_numbers) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", xrt_core::dlerror());
+      return;
+    }
+
+    using get_dtrace_buffer_size_fun = void (*)(uint64_t*);
+    auto get_dtrace_buffer_size = reinterpret_cast<get_dtrace_buffer_size_fun>
+        (xrt_core::dlsym(dtrace.lib_hdl.get(), "get_dtrace_buffer_size"));
     if (!get_dtrace_buffer_size) {
       xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", xrt_core::dlerror());
       return;
     }
 
-    using create_dtrace_buffer_fun = void (*)(const char*, const char*, size_t, uint64_t*, size_t, uint64_t*);
-    auto create_dtrace_buffer =
-        (create_dtrace_buffer_fun) xrt_core::dlsym(dtrace.lib_hdl.get(), "create_dtrace_buffer");
-    if (!create_dtrace_buffer) {
+    using populate_dtrace_buffer_fun = void (*)(uint32_t*, uint64_t);
+    auto populate_dtrace_buffer = reinterpret_cast<populate_dtrace_buffer_fun>
+       (xrt_core::dlsym(dtrace.lib_hdl.get(), "populate_dtrace_buffer"));
+    if (!populate_dtrace_buffer) {
       xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", xrt_core::dlerror());
       return;
     }
 
     // using function ptr get dtrace control buffer size
-    // this function also returns dtrace mem buffer size if mem read calls are used in control scipt.
-    uint32_t ctrl_buf_size;
-    uint32_t mem_buf_size;
-    if (get_dtrace_buffer_size(dtrace.ctrl_file_path.c_str(), dtrace.map_data.c_str(),
-                               &ctrl_buf_size, &mem_buf_size) != 0) {
+    uint32_t buffers_length = 0;
+    if (get_dtrace_col_numbers(dtrace.ctrl_file_path.c_str(), dtrace.map_data.c_str(),
+                               &buffers_length) != 0) {
       xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
-                              "[dtrace] : Failed to get control buffer size");
+                              "[dtrace] : Failed to get col numbers");
       return;
     }
 
-    // control buf size is empty, control script does nothing.
-    if (!ctrl_buf_size) {
+    if (!buffers_length) {
       xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
                               "[dtrace] : Control buffer size is zero, no dtrace o/p");
       return;
     }
 
+    // control buf size is empty, control script does nothing.
     // Create xrt::bo's with size returned and call the API to fill the buffers with
     // required data, buffers are synced after data is filled.
     try {
+      std::vector<uint64_t> buffers(buffers_length);
+      get_dtrace_buffer_size(buffers.data());
+
+      std::map<uint32_t, size_t> buf_sizes;
+      size_t total_size = 0;
+
+      const uint32_t MASK32 = 0xffffffff;
+      const uint32_t SHIFT32 = 32;
```

Comment:
Never ever ALLCAPS except for macros.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es9-avoid-all_caps-names

Plus these should be compile time constexpr

Reply:
sure. will change